sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -359,6 +359,14 @@ static bool getUint32PropAsBool(const RTDeviceBinaryImage &Img,
   return Prop && (DeviceBinaryProperty(Prop).asUint32() != 0);
 }
 
+static int getUint32PropAsInt(const RTDeviceBinaryImage &Img,
+                              const char *PropName) {
+  pi_device_binary_property Prop = Img.getProperty(PropName);
+  if (!Prop)
+    return -1;
+  return (int)(DeviceBinaryProperty(Prop).asUint32());
```

Comment:
i think we can remove the explicit cast but if we want to keep it i would use static_cast rather than c-style